client/client-lib/src/lib.rs

```diff
@@ -447,17 +447,21 @@ impl Client<UserClientConfig> {
         }
 
         // if db is empty, fetch from federation and save to db
-        let gateways = self.context.api.fetch_gateways().await?;
-        if gateways.is_empty() {
-            return Err(ClientError::NoGateways);
-        };
+        let gateways = self.select_gateways().await.unwrap();
         let gateway = gateways[0].clone();
         self.context
             .db
             .insert_entry(&LightningGatewayKey, &gateway)
             .expect("DB error");
         Ok(gateway)
     }
+    pub async fn select_gateways(&self) -> Result<Vec<LightningGateway>> {
+        let gateways = self.context.api.fetch_gateways().await?;
+        if gateways.is_empty() {
+            return Err(ClientError::NoGateways);
+        };
```

Comment:
I think for the cli command an empty list would be a valid outcome. I'm not sure if this function still makes sense without the error handling.

Reply:
Agreed. Just return empty array.

Reply:
Makes sense. I'm updating the behavior :)

As for having the function, I'm inspired by some of the future issues Justin hinted at and I'm thinking to explore dependency injection of the selection strategy so we have the client inject their own 'select-gateways' logic instead of the library being opinionated on how to select. A default strategy that can be afforded by the library might look like what we already have in `fetch_gateways`.

This is not necessary atm, but would such a pattern be useful in the larger scheme of gateways management?

Reply:
Yea I think users should be able to provide their own strategy for choosing a gateway. But like you said, that won't happen for a while.

Reply:
Oh, how would DI of a selection strategy  work in wasm?!

Reply:
I don't understand

Reply:
I'm wondering if mobile clients would be able to dependency inject a strategy to `fetch_gateway` function if they are taking a dependency on client-lib via wasm

Reply:
Haven't thought that far out yet

Reply:
well, I'll just put a 📌 on the idea and leave it at that for now :)

Reply:
> Let's delete this function

I think we should keep `select_gateways` and deprecate `fetch_gateway` as a public client-lib interface. A lot of the usages  of `fetch_gateway` only need one gateway and obviously I'm finding that migrating `fetch_gateway` to start returning a list now forces me to check for empty list case in every client-lib internal application of `fetch_gateway`

If I use this opportunity to reconsider public functionalities client-lib exposes for interacting with the gateways, I'd propose the following three clear functions
- `select_gateways(active?)` for listing all gateways registered on mint and in DB. We can build our gateway selection logic here in future. Private `fetch_gateway` within the client-lib should wrap `select_gateways`
- `activate_gateway(gateway)` for activating a gateway, i.e listing it as the first item in the DB?
- `deactivate_gateway(gateway)` for removing a gateway from the DB

Lmk if I should go ahead with the new functions to add capability